File: course_101/exercises/easy_7/swap_case.rb
# Write a method that takes a string as an argument and returns a new string in which every uppercase letter is replaced by its lowercase version,
# and every lowercase letter by its uppercase version.
# All other characters should be unchanged.
# You may not use String#swapcase; write your own version of this method.

# solution 1
def swapcase(string)
  swapped_case =  string.chars.map do |char|
    if char =~ /[a-z]/ then char.upcase
    elsif char =~ /[A-Z]/ then char.downcase
    else char.downcase
    end
  end
  swapped_case.join
end


puts swapcase('CamelCase') == 'cAMELcASE'
puts swapcase('Tonight on XYZ-TV') == 'tONIGHT ON xyz-tv'

File: course_js210_computational_thinking_and_javascript_programming/exercises/js210_small_problems/string_and_text_processing/swap_case.js
// Write a function that takes a string as an argument, and returns that string with every lowercase letter changed to uppercase and every uppercase letter changed to lowercase. Leave all other characters unchanged.

function swapCase(string) {
  return string.split('').map(function (letter) {
    if (/[a-z]/.test(letter)) {
      return letter.toUpperCase();
    } else if (/[A-Z]/.test(letter)) {
      return letter.toLowerCase();
    } else return letter;
  }).join('');
}

console.log(swapCase('CamelCase'));              // "cAMELcASE"
console.log(swapCase('Tonight on XYZ-TV'));      // "tONIGHT ON xyz-tv"
